### PR TITLE
✨ Add retryProcessInstance

### PR DESCRIPTION
**Pull request by alexanderkasten**
## Changes

1. Add retryProcessInstance

## Issues

Part of https://github.com/atlas-engine/AtlasEngine.FullstackServer/issues/12

PR: #2 

## How to test the changes

> Describe how others can test your changes (not required when fixing typos, linter errors and such)

### CODE REVIEW

File: src/rest_settings.ts [hunk 1]
```diff
@@ -37,6 +37,12 @@ const paths = {
    * @method delete
    */
   terminateProcessInstance: `/process_models/${params.processInstanceId}/terminate`,
+  /*
+   * Sends a signal to terminate the ProcessInstance with the given ID.
```
Comment (rrrene) [inline src/rest_settings.ts]:
```suggestion
   * Sends a signal to terminate the ProcessInstance with the given ID.

```

File: src/adapters/iprocess_model_adapter.ts [hunk 2]
```diff
@@ -45,4 +45,18 @@ export interface IProcessModelAdapter {
     processInstanceId: string,
     identity?: IIdentity,
   ): Promise<void>;
+
+  /**
+   * Retries a ProcessInstance.
+   *
+   * @async
+   * @param  processInstanceId   The ID of the ProcessInstance to terminate.
```
Comment (ElRaptorus) [inline src/adapters/iprocess_model_adapter.ts]:
```suggestion
   * @param  processInstanceId   The ID of the ProcessInstance to retry.
```

File: src/repositories/iflow_node_instance_repository.ts [hunk 3]
```diff
@@ -91,4 +91,16 @@ export interface IFlowNodeInstanceRepository {
     offset?: number,
     limit?: number,
   ): Promise<Array<FlowNodeInstance>>;
+
+  /**
+   * Resets FlowNodeInstance's state to running and deletes its token except the onEnter token.
```
Comment (ElRaptorus) [inline src/repositories/iflow_node_instance_repository.ts]:
```suggestion
   * Resets the state of the given FlowNodeInstance to running.
```

File: src/repositories/iflow_node_instance_repository.ts [hunk 4]
```diff
@@ -91,4 +91,16 @@ export interface IFlowNodeInstanceRepository {
...
+   *
+   * This only works on FlowNodeInstances, which have previously been put
+   * into a 'error' state.
```
Comment (ElRaptorus) [inline src/repositories/iflow_node_instance_repository.ts]: Solche linebreaks bitte vermeiden. Dieses Nur-80-Zeichen-Pro-Zeile-Dogma gibts nicht mehr. :wink: 

```suggestion 
   * Works only on FlowNodeInstances that are in an error state.
```

File: src/repositories/iprocess_instance_repository.ts [hunk 5]
```diff
@@ -69,4 +69,16 @@ export interface IProcessInstanceRepository {
    *                            ProcessInstance.
    */
   terminate(identity: IIdentity, processInstanceId: string, terminatedByUserId?: string): Promise<void>;
+
+  /**
+   * Resets the given ProcessInstance to a running state.
```
Comment (ElRaptorus) [inline src/repositories/iprocess_instance_repository.ts]:
```suggestion
   * Resets the state of the given ProcessInstance to running.
   * Works only on ProcessInstances that are in an error state.
```

File: src/rest_settings.ts [hunk 6]
```diff
@@ -37,6 +37,13 @@ const paths = {
...
```
Comment (ElRaptorus) [inline src/rest_settings.ts]:
```suggestion
   * Retries the ProcessInstance with the given ID. This only works for ProcessInstances that have finished with an error.
```

File: src/rest_settings.ts [hunk 7]
```diff
@@ -37,6 +37,13 @@ const paths = {
...
+
+   * @tag ProcessInstances
+   * @method delete
```
Comment (ElRaptorus) [inline src/rest_settings.ts]:
```suggestion
   * @method put
```

File: src/engine/services/iresume_process_service.ts [hunk 8]
```diff
@@ -26,4 +26,14 @@ export interface IResumeProcessService {
      * @returns                   The ProcessInstance's result or nothing, if the ProcessInstance was already finished.
      */
   resumeProcessInstanceById(identity: IIdentity, processModelId: string, processInstanceId: string): Promise<BpmnEventMessage | void>;
+
+  /**
+   *  Retries a specific ProcessInstance by its ID.
+   *
+   *  This does only work with ProcessInstances which are in an error state.
```
Comment (ElRaptorus) [inline src/engine/services/iresume_process_service.ts]:
```suggestion
   *  Works only on ProcessInstances that have finished with an error.
```

**Pull request merged**